sdk/appconfiguration/app-configuration/src/internal/shims.d.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// d.ts shims provide types for things we use internally but are not part
+// of @azure/app-configuration's surface area.
```

Comment:
They came from me just browsing to definition in a TS project that used them - so mostly from the lib.es5.d.ts files that are built-in.

If there is a better way I'm all for it - it was a pain and I don't like having to maintain them - this was the technique used in core-amqp but that might have been used because what it depended on was so small.

Reply:
Doing a quick search gives me
- https://www.npmjs.com/package/@types/text-encoding for TextEncoder
- https://www.npmjs.com/package/@types/btoa for btoa
- https://www.npmjs.com/package/@types/atob for atob


As mentioned in the other comment, we shouldnt need to add anything for Uint8Array and ArrayBuffer.

Reply:
Fixed, although differently.

Apparently the guidelines recommend just `/// <reference`ing the dom (rather that including it in tsconfig, which would be the alternative). So I've done that and now we don't have to have a shim at all.